### PR TITLE
Fix up attestation pool

### PR DESCRIPTION
**Pull request by terencechain**
The current attestation pool needs structure changes to better support:
1.) filter out seen attestations from the wire 
2.) attestations to block inclusions 

Change list:
- Updated aggregated attestation pool to `map[HTR(att.Data)]*[]ethpb.Attestation` from `map[HTR(att)]*ethpb.Attestation`
- Updated `HasAggregatedAttestation` to consider bitfield containment 
- Better tests

### CODE REVIEW

File: beacon-chain/operations/attestations/kv/aggregated.go [hunk 1]
```diff
@@ -13,15 +13,34 @@ func (p *AttCaches) SaveAggregatedAttestation(att *ethpb.Attestation) error {
...
-
-	r, err := ssz.HashTreeRoot(att)
+	r, err := ssz.HashTreeRoot(att.Data)
 	if err != nil {
 		return errors.Wrap(err, "could not tree hash attestation")
 	}
 
+	key := string(r[:])
+	var atts []*ethpb.Attestation
+	d, ok := p.aggregatedAtt.Get(key)
```
Comment (rauljordan) [inline beacon-chain/operations/attestations/kv/aggregated.go]: doing `p.aggregatedAtt.Get(string(r[:]))` is actually faster underneath the hood in Go: https://github.com/golang/go/issues/3512, at least for maps (small tidbit)

File: beacon-chain/operations/attestations/kv/aggregated_test.go [hunk 2]
```diff
@@ -108,43 +108,174 @@ func TestKV_Aggregated_CheckExpTime(t *testing.T) {
...
+			},
+			input: &ethpb.Attestation{
+				Data: &ethpb.AttestationData{
+					Slot: 1,
+				},
+				AggregationBits: bitfield.Bitlist{0b1111}},
+			want: false,
+		},
+		{
+			name: "multiple attestations with same data in cache with overlapping aggregation, input is subset",
```
Comment (rauljordan) [inline beacon-chain/operations/attestations/kv/aggregated_test.go]: love these tests

File: beacon-chain/operations/attestations/kv/block.go [hunk 3]
```diff
@@ -9,14 +9,34 @@ import (
...
 func (p *AttCaches) SaveBlockAttestation(att *ethpb.Attestation) error {
-	r, err := ssz.HashTreeRoot(att)
+	r, err := ssz.HashTreeRoot(att.Data)
 	if err != nil {
 		return errors.Wrap(err, "could not tree hash attestation")
 	}
 
+	key := string(r[:])
+	var atts []*ethpb.Attestation
+	d, ok := p.blockAtt.Get(key)
```
Comment (rauljordan) [inline beacon-chain/operations/attestations/kv/block.go]: prefer the `p.blockAtt.Get(string(r[:]))` lookup

**Comment by prestonvanloon** [comment]
Tested in experimental, showing significant improvement to gRPC latency for most hard hitting services.

**Pull request merged**